clouddrift/plotting.py

```diff
@@ -0,0 +1,212 @@
+
+
+def plot_ragged(
+    ax,
+    longitude: Union[list, np.ndarray, pd.Series, xr.DataArray],
+    latitude: Union[list, np.ndarray, pd.Series, xr.DataArray],
+    rowsize: Union[list, np.ndarray, pd.Series, xr.DataArray],
+    *args,
+    colors: Optional[Union[list, np.ndarray, pd.Series, xr.DataArray]] = None,
+    tolerance: Optional[Union[float, int]] = 180,
+    **kwargs,
+):
+    """Plot trajectories from a ragged array dataset on a Matplotlib Axes
+    or a Cartopy GeoAxes object ``ax``.
+
+    This function wraps Matplotlib's ``plot`` function (``plt.plot``) and
+    ``LineCollection`` (``matplotlib.collections``) to efficiently plot
+    trajectories from a ragged array dataset.
+
+    Parameters
+    ----------
+    ax: matplotlib.axes.Axes or cartopy.mpl.geoaxes.GeoAxes
+        Axis to plot on.
+    longitude : array-like
+        Longitude sequence. Unidimensional array input.
+    latitude : array-like
+        Latitude sequence. Unidimensional array input.
+    rowsize : list
+        List of integers specifying the number of data points in each row.
+    *args : tuple
+        Additional arguments to pass to ``ax.plot``.
+    colors : array-like
+        Colors to use for plotting. If colors is the same shape as longitude and latitude,
+        the trajectories are splitted into segments and each segment is colored according
+        to the corresponding color value. If colors is the same shape as rowsize, the
+        trajectories are uniformly colored according to the corresponding color value.
+    tolerance : float
+        Longitude tolerance gap between data points (in degrees) for segmenting trajectories.
+        For periodic domains, the tolerance parameter should be set to the maximum allowed gap
+        between data points. Defaults to 180.
+    **kwargs : dict
+        Additional keyword arguments to pass to ``ax.plot``.
+
+    Returns
+    -------
+    list of matplotlib.lines.Line2D or matplotlib.collections.LineCollection
+        The plotted lines or line collection. Can be used to set a colorbar
+        after plotting or extract information from the lines.
+
+    Examples
+    --------
+
+    Plot the first 100 trajectories from the gdp1h dataset, assigning
+    a different color to each trajectory:
+
+    >>> from clouddrift import datasets
+    >>> ds = datasets.gdp1h()
+    >>> ds = subset(ds, {"ID": ds.ID[:100].values}).load()
+    >>> fig = plt.figure()
+    >>> ax = fig.add_subplot(1, 1, 1)
+
+    >>> plot_ragged(
+    >>>     ax,
+    >>>     ds.lon,
+    >>>     ds.lat,
+    >>>     ds.rowsize,
+    >>>     colors=np.arange(len(ds.rowsize))
+    >>> )
+
+    To plot the same trajectories, but assigning a different color to each
+    observation and specifying a colormap:
+
+    >>> fig = plt.figure()
+    >>> ax = fig.add_subplot(1, 1, 1)
+    >>> lc = plot_ragged(
+    >>>     ax,
+    >>>     ds.lon,
+    >>>     ds.lat,
+    >>>     ds.rowsize,
+    >>>     colors=np.floor(time),
```

Comment:
`time` is not defined for this example. @philippemiron what did you intend it to be?